backend/sqlite/sqlite.go

```diff
@@ -390,20 +392,57 @@ func (be *sqliteBackend) CompleteOrchestrationWorkItem(ctx context.Context, wi *
 }
 
 // CreateOrchestrationInstance implements backend.Backend
-func (be *sqliteBackend) CreateOrchestrationInstance(ctx context.Context, e *backend.HistoryEvent) error {
+func (be *sqliteBackend) CreateOrchestrationInstance(ctx context.Context, e *backend.HistoryEvent, option *protos.CreateInstanceOption) error {
 	if err := be.ensureDB(); err != nil {
 		return err
 	}
 
+	// if user not set this option, it will be nil, initialize it here to set default value
+	if option == nil {
+		option = &protos.CreateInstanceOption{}
+	}
+
 	tx, err := be.db.BeginTx(ctx, nil)
 	if err != nil {
 		return fmt.Errorf("failed to start transaction: %w", err)
 	}
 	defer tx.Rollback()
 
 	var instanceID string
-	if err := be.createOrchestrationInstanceInternal(ctx, e, tx, &instanceID); err != nil {
-		return err
+	runtimeStatus, instanceID, err := be.createOrchestrationInstanceInternal(ctx, e, tx)
+	if err != nil {
+		// instance alredy exists
+		if errors.Is(err, backend.ErrDuplicateEvent) {
```

Comment:
Why not move this logic directly into `createOrchestrationInstanceInternal`? It seems that it would be valuable to support this for both top-level orchestrations and sub-orchestrations. It would also simplify the code if we didn't have to return so many values.

We don't necessarily have to add API support for sub-orchestrations, but it would be nice if the core implementation is ready to support it.